torch/distributed/_shard/sharded_tensor/_ops/linear.py

```diff
@@ -141,13 +141,13 @@ def _validate_linear_op_param(args, kwargs):
     # Validate types
     if not isinstance(input, torch.Tensor) and not isinstance(input, ShardedTensor):
         raise TypeError("input needs to be either torch.Tensor or ShardedTensor")
-    if not isinstance(bias, torch.Tensor):
+    if isinstance(bias, ShardedTensor):
```

Comment:
Nit: this may not needed right?

Reply:
Yeah I think we can use `issubclass` instead, let me update it.

Reply:
Actually this is tricker, because `bias` can either be a `torch.Tensor` or `torch.nn.Parameter`. so a way to check an object is torch.Tensor and not its subclass will fail in the case when `bias` is a `torch.nn.Parameter`. 

So a robust way might be `type(bias) != torch.Tensor and type(bias) != nn.Parameter`, which is a bit nasty but I guess we could do that for now